targets/TARGET_STM/TARGET_STM32F7/device/stm32f7xx_hal_pcd.c

```diff
@@ -1334,7 +1334,7 @@ static HAL_StatusTypeDef PCD_WriteEmptyTxFifo(PCD_HandleTypeDef *hpcd, uint32_t
     ep->xfer_count += len;
   }
 
-  if(len <= 0)
+  if ((int32_t)len <= 0)
```

Comment:
keep original code

Reply:
@jeromecoutant   No, I don't think that'll work.

Since `len` is an uint32_t now, it cannot be less than zero.

That is why I added checks for `(int32_t)len  > 0)` because if `ep->xfer_count` is greater than `ep->xfer_len` then `len` would have its high bit set, and enters the scope that sets `len = ep->maxpacket;` which is incorrect.  The high bit should be preserved so that `atomic_clr_u32` is called as expected.

If I make those changes as you suggest, that would change the behavior of the code.

Is my understanding/interpretation wrong?